teraserver/python/modules/FlaskModule/API/user/UserQueryDeviceSubTypes.py

```diff
@@ -47,31 +47,44 @@ def get(self):
         parser = get_parser
 
         args = parser.parse_args()
+        has_list = args.pop('list')
 
         device_subtypes = []
-        # If we have no arguments, return all accessible devices
         # If we have no arguments, return error
         if not any(args.values()):
             return gettext('Missing arguments'), 400
 
-        if args['id_device_subtype']:
-            device_subtypes = TeraDeviceSubType.query.filter(TeraDeviceSubType.id_device_type.
-                                                             in_(user_access.get_accessible_devices_types_ids())).\
-                filter_by(id_device_subtype=args['id_device_subtype']).all()
+        #if we have 2 IDs, return error
+        elif args['id_device_subtype'] is not None and args['id_device_type'] is not None:
+            return gettext('Too Many IDs'), 400
+
+        elif args['id_device_subtype']:
+            if args['id_device_subtype'] in user_access.get_accessible_devices_subtypes_ids():
+                device_subtypes = TeraDeviceSubType.query.filter(TeraDeviceSubType.id_device_type.
+                                                                 in_(user_access.get_accessible_devices_types_ids())).\
+                    filter_by(id_device_subtype=args['id_device_subtype']).all()
+            else:
+                return gettext('No access to device subtype'), 403
         elif args['id_device_type']:
             # Check if has access to the id_device_type
-            if not args['id_device_type'] in user_access.get_accessible_devices_types_ids():
-                return 'No access to device type', 403
-            device_subtypes = TeraDeviceSubType.get_device_subtypes_for_type(args['id_device_type'])
-        elif args['list']:
-            device_subtypes = user_access.get_accessible_devices_subtypes()
+            if args['id_device_type'] in user_access.get_accessible_devices_types_ids():
+                device_subtypes = TeraDeviceSubType.get_device_subtypes_for_type(args['id_device_type'])
+            else:
+                return gettext('No access to device type'), 403
+
+        else:
+            return gettext('ID can\'t be 0'), 400
 
         try:
             device_subtypes_list = []
             for dst in device_subtypes:
                 if dst is not None:
-                    dst_json = dst.to_json()
-                    device_subtypes_list.append(dst_json)
+                    if has_list:
+                        dst_json = dst.to_json(minimal=True)
+                        device_subtypes_list.append(dst_json)
+                    else:
+                        dst_json = dst.to_json()
+                        device_subtypes_list.append(dst_json)
```

Comment:
This could probably simplified without using "if":
dst_json=dst.to_json(minimal=has_list)

To be tested!

Reply:
Will do! Thank you for the suggestion